docs/src/your-own-sender.md

```diff
@@ -0,0 +1,197 @@
+
+Next we add a simple function used to construct the sender:
+```cpp
+write_sender write(uv_stream_t *handle, const uv_buf_t *bufs, size_t nbufs) {
+	return write_sender{handle, bufs, nbufs};
+}
+```
+
+In addition to that, we need the `connect` overload:
+```cpp
+template <typename Receiver>
+write_operation<Receiver> connect(write_sender s, Receiver r) {
+	return {s, std::move(r)};
+}
+```
+
+`connect` simply constructs an operation from the sender and receiver.
+
+We also add an implementation of `operator co_await` for our class so that we can `co_await` it inside of a coroutine:
+```cpp
+async::sender_awaiter<write_sender, write_sender::value_type>
+operator co_await(write_sender s) {
+	return {s};
+}
+```
+
+`async::sender_awaiter` is a special type that can suspend and resume the coroutine, and internally connects a receiver to our sender.
+
+### The operation
+
+With the sender done, what remains to be written is the operation. As noted earlier, the operation is constructed using the sender and receiver, and it stores the operation state. Let's start by writing a skeleton for the class:
+```cpp
+template <typename Receiver>
+struct write_operation {
+	write_operation(write_sender s, Receiver r)
+	: req_{}, handle_{s.handle}, bufs_{s.bufs}, nbufs_{s.nbufs}, r_{std::move(r)} { }
+
+private:
+	uv_write_t req_;
+	uv_stream_t *handle_;
+	const uv_buf_t *bufs_;
+	size_t nbufs_;
+
+	Receiver r_;
+};
+```
+
+The operation stores all the necessary state, and is templated on the receiver type in order to support any receiver type.
+
+Next, we add a `start_inline` method:
+```cpp
+bool start_inline() {
+	auto result = uv_write(&req_, handle_, bufs_, nbufs_, [] (uv_write_t *req, int status) {
+		/* TODO */
+	});
+
+	if (result < 0) {
+		async::execution::set_value_inline(r_, result);
+		return true; // Completed inline
+	}
+
+	return false; // Did not complete inline
+}
+```
+
+We use `start_inline` here in order to notify the user of any immediate errors synchronously. We use functions defined inside of `async::execution` to set the value, because they properly detect which method should be called on the receiver.
+
+Now, let's implement the actual asynchronous completion:
+```cpp
+...
+	handle_->data = this;
+	auto result = uv_write(&req_, handle_, bufs_, nbufs_, [] (uv_write_t *req, int status) {
+		auto op = static_cast<write_operation *>(req->handle->data);
+		op->complete(status);
+	});
+...
+```
+
+We use the handle's user data field to store a pointer to our operation in order to be able to access it later. Do note that this way of implementing it means that only one write operation may be in progress at once. One way to solve this would be to have a wrapper object that manages the handle and has a map of `req` to `write_operation`, but that's beyond the scope of this example.
```

Comment:
maybe the reader: why? wouldn't a different sender/receiver/operation bunch be generated for each call to write?

Reply:
I've clarified it to mention that the pointer points to the instance of the operation associated with this invocation of `write`.